### PR TITLE
IGNITE-19103: Sql. Change implicit primary key column's type to UUID.

### PR DESCRIPTION
**Pull request by lowka**
Changes type of implicit PK to UUID.
Removes `gen_random_uuid` function.

https://issues.apache.org/jira/browse/IGNITE-19103

---

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/schema/src/main/java/org/apache/ignite/internal/schema/DefaultValueGenerator.java [hunk 1]
```diff
@@ -27,8 +27,8 @@
  * (i.e. could be specified as a default in column definition).
  */
 public enum DefaultValueGenerator {
-    /** Generator that generates random UUID string. */
-    GEN_RANDOM_UUID("genRandomUuid", String.class);
+    /** Generator that generates random UUID. */
+    RAND_UUID("randUuid", String.class);
```
Comment (AMashenkov) [inline modules/schema/src/main/java/org/apache/ignite/internal/schema/DefaultValueGenerator.java:31]:
```suggestion
    RAND_UUID("randUuid", UUID.class);
```

File: modules/schema/src/main/java/org/apache/ignite/internal/schema/DefaultValueGenerator.java [hunk 2]
```diff
@@ -56,4 +56,10 @@ public Object next() {
     public static String genRandomUuid() {
         return UUID.randomUUID().toString();
     }
+
+    /** Returns random UUID string. */
+    @SuppressWarnings("unused") // actually method is called via reflection
+    public static String randUuid() {
+        return UUID.randomUUID().toString();
```
Comment (AMashenkov) [inline modules/schema/src/main/java/org/apache/ignite/internal/schema/DefaultValueGenerator.java:63]:
```suggestion
    /** Returns random UUID string. */
    @SuppressWarnings("unused") // actually method is called via reflection
    public static UUID randUuid() {
        return UUID.randomUUID();
```

Comment (lowka) [reply modules/schema/src/main/java/org/apache/ignite/internal/schema/DefaultValueGenerator.java]: Thanks fixed.

**Review by AMashenkov — APPROVED** [review]

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java [hunk 3]
```diff
@@ -593,28 +593,28 @@ public static HybridTimestamp clusterWideEnsuredActivationTsSafeForRoReads(
...
-     */
-    static boolean isSupportedFunctionalDefault(String functionName) {
-        return FUNCTIONAL_DEFAULT_FUNCTIONS.contains(functionName.toUpperCase());
-    }
-
     /**
      * Check if provided default value is a constant or a functional default of supported function, or fail otherwise.
      */
-    static void ensureSupportedDefault(String columnName, @Nullable DefaultValue defaultValue) {
+    static void ensureSupportedDefault(String columnName, @Nullable DefaultValue defaultValue, ColumnType columnType) {
```
Comment (xtern) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java:599]:
```suggestion
    static void ensureSupportedDefault(String columnName, ColumnType columnType, @Nullable DefaultValue defaultValue) {
```
to simplify future merge

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/IgniteMethod.java [hunk 4]
```diff
@@ -98,17 +98,13 @@ public enum IgniteMethod {
...
+    RAND_UUID(IgniteSqlFunctions.class, "randUuid"),
 
     LENGTH(IgniteSqlFunctions.class, "length", Object.class),
 
     OCTET_LENGTH(IgniteSqlFunctions.class, "octetLength", ByteString.class),
     OCTET_LENGTH2(IgniteSqlFunctions.class, "octetLength", String.class),
 
-    /** See {@link IgniteSqlFunctions#genRandomUuid()}. */
-    // TODO This function should removed when https://issues.apache.org/jira/browse/IGNITE-19103 is complete.
-    GEN_RANDOM_UUID(IgniteSqlFunctions.class, "genRandomUuid"),
```
Comment (xtern) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/IgniteMethod.java:110]: please remove the function body from `IgniteSqlFunctions` class as well.

Comment (lowka) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/IgniteMethod.java:110]: Fixed.

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java (hunk 3 above)
Comment (lowka) [reply modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java]: Fixed.

**Review by xtern — APPROVED** [review]

**Pull request merged**